katsdpcontroller/generator.py

```diff
@@ -1718,29 +1737,31 @@ def _render_continuum_parameters(parameters):
     return '; '.join('{}={!r}'.format(key, value) for key, value in parameters.items())
 
 
-def _make_continuum_imager(g, config, capture_block_id, name, telstate, target_cache):
+def _make_continuum_imager(g, config, capture_block_id, name, telstate, target_mapper):
     output = config['outputs'][name]
     l1_flags_name = output['src_streams'][0]
     l0_name = config['outputs'][l1_flags_name]['src_streams'][0]
     l0_info = L0Info(config, l0_name)
     l0_stream = name + '.' + l0_name
     data_url = _stream_url(capture_block_id, l0_stream)
     cpus = _continuum_imager_cpus(config)
-    targets = _unique_targets(config, capture_block_id, name, telstate, target_cache)
+    # Image targets observed for at least 15 minutes
+    targets = _get_targets(config, capture_block_id, name, telstate, 15 * 60)
 
     # katdal doesn't support selection by target description, and selection by
     # name can be ambiguous (if there are multiple targets with the same
     # name), so we select by index.
-    catalogue = _get_targets(telstate, capture_block_id, l0_stream)
+    catalogue = _get_data_set(telstate, capture_block_id, l0_stream).catalogue
```

Comment:
Do we really need to do this again ? _get_targets has just trawled the catalogue and picked targets from it - how would they disappear between these calls ?

Reply:
Yeah, I wasn't totally happy with this either. I think I might just fix https://skaafrica.atlassian.net/browse/SR-1819, after which it won't be necessary to get the catalogue again to look up indices.

Reply:
Or we could return the indices together with the target mapping from the _get_data_set call

Reply:
That was my fallback idea too.